macros/commands/enforce_project_configurations.sql

```diff
@@ -0,0 +1,66 @@
+{%- macro enforce_project_configurations(enforce_owners=true, enforce_tags=false, enforce_meta_params=[], enforce_config_params=[], exclude_sources=false) -%}
+    {%- if execute -%}
+        {# enforcing source params #}
+        {%- if not exclude_sources -%}
+            {% set sources = graph.sources.values() | selectattr('resource_type', '==', 'source') %}
+            {% set sources_result = elementary.enforce_configuration(sources, elementary.flatten_source, enforce_owners, enforce_tags, enforce_meta_params, enforce_config_params) %}
+        {%- endif -%}
+
+        {# enforcing model params #}
+        {% set models = graph.nodes.values() | selectattr('resource_type', '==', 'model') %}
+        {% set models_result = elementary.enforce_configuration(models, elementary.flatten_model, enforce_owners, enforce_tags, enforce_meta_params, enforce_config_params) %}
+
+        {%- if not models_result or not sources_result -%}
+            {{ exceptions.raise_compiler_error("Found issues in projdct configurations") }}
+        {%- endif -%}
+    {%- endif -%}
+{%- endmacro -%}
+
+{%- macro get_enforcement_param(flattened_node, enforcement_param_name, enforcement_param_arg_value) -%}
+    {% set node_enforcement_param = flattened_node.meta.get(enforcement_param_name) or flattened_node.config.get(enforcement_param_name) or elementary.get_config_var(enforcement_param_name) or enforcement_param_arg_value %}
+    {{- return(node_enforcement_param) -}}
+{%- endmacro -%}
+
+{%- macro enforce_configuration(nodes, flatten_callback, enforce_owners, enforce_tags, enforce_meta_params, enforce_config_params) -%}
+    {% set validation_result = {'success': true} %}
+    {% for node in nodes -%}
+        {% set flattened_node = flatten_callback(node) %}
+        {%- if flattened_node.package_name == project_name -%}
```

Comment:
What do you think about making this at the node-fetching level (`{% set models = ... %}`)?